contrib/mesos/pkg/executor/executor.go

```diff
@@ -206,10 +206,23 @@ func (k *Executor) isDone() bool {
 }
 
 // sendPodUpdate assumes that caller is holding state lock; returns true when update is sent otherwise false
-func (k *Executor) sendPodUpdate(u *kubetypes.PodUpdate) bool {
+func (k *Executor) sendPodUpdate(sendSnapshot bool) bool {
 	if k.isDone() {
 		return false
 	}
+	var snapshot []*api.Pod
+	if sendSnapshot {
+		snapshot = make([]*api.Pod, 0, len(k.pods))
```

Comment:
we send snapshots (SET operations) because that what the new merged source deals with. I didn't feel like copy/pasting logic everywhere to convert the pods map into a slice, so I centralized that here. I can rename the func to something like `sendPodsSnapshot` if you think that's more clear.